lib/Doctrine/Common/Collections/ArrayCollection.php

```diff
@@ -325,12 +325,12 @@ public function getIterator() : Traversable
      * @return static
      *
      * @psalm-template U
-     * @psalm-param Closure(T=):U $func
+     * @psalm-param Closure(T=, TKey):U $func
      * @psalm-return static<TKey, U>
      */
     public function map(Closure $func) : Collection
     {
-        return $this->createFrom(array_map($func, $this->elements));
+        return $this->createFrom(array_map($func, $this->elements, array_keys($this->elements)));
```

Comment:
Seems very much valid to me, but requires test additions to prevent regressions later on.

Reply:
Oh I forgot to add it. Thanks :+1:

Reply:
The tests are written in [PHPUnit](https://phpunit.readthedocs.io/en/8.4/) and can be found in `tests/Doctrine/Tests`. You will find that most of the time there is one test class that has the same name as the class you have changed, with the difference of a suffix called "Test".

In your case you have changed code in `lib/Doctrine/Common/Collections/ArrayCollection.php` and you'll find a file called `tests/Doctrine/Tests/Common/Collections/ArrayCollectionTest.php` that contains test methods. Because of the several types of collections which have to behave the same, it extends a `BaseArrayCollectionTest` class. Other test classes are extending `BaseCollectionTest`.

You'll have to create your new test method for your changes into the `BaseArrayCollectionTest` and `BaseCollectionTest`. You can take a look at `BaseCollectionTest::testMap()` to see how such a test can look like.

@Ocramius: Isn't `BaseArrayCollectionTest` supposed to extend `BaseCollectionTest`?